securedrop/tests/test_journalist_api.py

```diff
@@ -53,6 +56,15 @@ def test_valid_user_can_get_an_api_token(journalist_app, test_journo):
         assert response.json['journalist_first_name'] == test_journo['first_name']
         assert response.json['journalist_last_name'] == test_journo['last_name']
 
+        def _valid_date(date_str, date_format):
```

Comment:
Maybe I'm missing something, is there any reason this couldn't just be:
```python
def _valid_date(date_str, date_format):
    assert date_str == datetime.strptime(date_str, date_format).strftime(date_format)
```

If the assertion fails, pytest will show us the values of the left and right sides instead of something generic like "asserting False is True". And we let the ValueError bubble up since it'll fail the test anyways.